### DISCUSSION TITLE
Mastodon Support for OAuth in Pocketbase ?

### DISCUSSION THREAD

**Discussion by benoitpetit** [pocketbase/pocketbase#2361] [Q&A]
Hello,

As an enthusiastic user of Pocketbase, I greatly appreciate the features it offers. However, I notice that Pocketbase currently does not support Mastodon authentication for OAuth. I would like to know if this is planned for a future update.

I believe that the addition of Mastodon authentication for OAuth would be very helpful, as it would make it easier for users who prefer to connect via Mastodon to use Pocketbase.

[https://joinmastodon.org/](https://joinmastodon.org/)

I am curious to know if Pocketbase plans to add this feature in the near future. If someone could provide me with information on this subject, that would be great.

Thank you very much! 😄

**Comment by ganigeorgiev** [comment]
I'm not planning working on Mastodon integration, but user contributions are welcomed.

I cannot find in their documentation, but it seems that OIDC is supported (https://github.com/mastodon/mastodon/pull/16221) and if that works then you should be able to use the existing OpenID Connect provider (_I'm not sure if this is only for login in the platform and whether it is exposed in their [OAuth2 apis](https://docs.joinmastodon.org/spec/oauth/)_).

Alternatively, a dedicated Mastodon provider could be also implemented. As a reference, you can use for example the [Gitea PR](https://github.com/pocketbase/pocketbase/pull/1643/files) (_you can ignore the `ui/dist` files_).

**Comment by mapperfr** [comment]
I would be very interested by a dedicated Mastodon provider as well, though the added complexity regarding Mastodon would probably be the federated architecture and the need to specify the instance to authenticate from. I'm not a Go dev myself but hopefully someone will seize the challenge :)
